tree/tree.h

```diff
@@ -0,0 +1,47 @@
+#ifndef _TREE_H__ 
+#define _TREE_H__
+
+#include "splitter.h"
+
+#include <vector>
+
+namespace tree {
+
+/**
+ * Base class for all Decision Tree implementations.
+ */
+class Tree{
+
+  public:
+
+    /**
+     * \brief Standard destructor.
+     */
+    virtual ~Tree() { }
+
+    /**
+     * Builds the Decision Tree
+     * \param X              An object of Collection
+     * \param sample_weight  Weight of each sample in collection
+     * \param feature_weight Weight of each feature 
```

Comment:
It was a @gcmgomes suggestion to use \ instead of @. I particularly prefer because differ java from c/c++ comment somehow.
There's no need for more elaborate description since the functionality of the method is pretty obvious (method's name is self-explainable). Secondly, \brief is optional.